packages/components/pie-modal/test/component/pie-modal.spec.ts

```diff
@@ -83,10 +93,162 @@ test.describe('`Pie Modal is closed`', () => {
                 },
             );
 
+            // Act
             await page.locator('#dialog').click();
 
+            // Assert
             expect(messages).toHaveLength(1);
         });
     });
 });
 
+test.describe('Pie Modal `isDismissible`', () => {
+    test.describe('when `isDismissible` is truthy', () => {
+        test('should contain a close button so the modal can be dismissed', async ({ mount }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: true,
+                    },
+                },
+            );
+
+            // Act & Assert
+            await expect(component.locator('.c-modal-closeBtn')).toBeVisible();
+        });
+
+        test('should close the modal when the close button is clicked', async ({ mount }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: true,
+                    },
+                },
+            );
+
+            // Act
+            await component.locator('[data-test-id="c-modal-closeBtn"]').click();
+
+            // Assert
+            await expect(component).not.toBeVisible();
+        });
+
+        test('should close the modal when the backdrop is clicked', async ({ mount, page }) => {
+            // Arrange
+            await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: true,
+                    },
+                },
+            );
+
+            // Act
+            await page.locator('body').click();
+
+            const element = await page.locator('#dialog');
+
+            const styles = await element.evaluate((modal) => {
+                const computedStyles = window.getComputedStyle(modal);
+                return {
+                    display: computedStyles.getPropertyValue('display'),
+                };
+            });
+
+            // Assert
+            expect(styles.display).toBe('none');
+        });
+
+        test('should close the modal when the ESC key is triggered', async ({ mount, page }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: false,
+                    },
+                },
+            );
+
+            // Act
+            await page.keyboard.press('Escape');
+
+            // Assert
+            await expect(component).not.toBeVisible();
```

Comment:
Is there a reason we are using two different methods of asserting the modal is closed in this and the test above?

Reply:
Yup. The backdrop is a bit of an issue to test. This one is fine because we can test it easily so we can use the `toBeVisible` helper. This helper does not work with the backdrop test above.